### ISSUE TITLE
Question about the config file

### ISSUE THREAD

**Issue by AhmedAtefAhmedAly** [xmed-lab/CardiacCLIP#2]
Hello, 

I have a question regarding replication of the results. I ran the script as instructed and got this result for the model with the lowest validation loss. For the 1-shot experiment, I am getting an mae of 11.3.  

`{"loss": 20.374399185180664, "ce_loss": 3.079083204269409, "kl_loss": 6.0367608070373535, "reg_loss": 11.25855541229248, "mae_exp_metric": 11.25855541229248, "acc_exp_metric": 0.0, "mae_max_metric": 11.25855541229248, "acc_max_metric": 0.0, "epoch": 0, "output_dir": "results/MICCAI/few-shot/1-shot/5e-5/version_9", "ckpt_path": "results/MICCAI/few-shot/1-shot/5e-5/version_9/ckpts/epoch=00-val_mae_exp_metric=11.3243.ckpt"}
`

I noticed that is because the config for the results is different than the default.yaml. However, the config does not work with the rest of the repository.

**Comment by duyao-art** [comment]
Hi, sorry for the late response.

May I ask which configuration file you used to run the experiments — the config files under results/MICCAI/few-shot/ for each shot setting, or the default default.yaml?

In our paper, all few-shot experiments were conducted using the former configuration setup (i.e., the shot-specific configs in results/MICCAI/few-shot/), rather than default.yaml.

Additionally, when you mentioned that “the config does not work with the rest of the repository”, could you please clarify what you mean by that? For example, did you encounter runtime errors, incompatible arguments, or unexpected behavior during training or evaluation?

**Comment by AhmedAtefAhmedAly** [comment]
Hello, 

Thanks for your response.

The config for example assigns weight to, rank loss and dist loss. So when you run the script with the config it raises a runtime error.  
    `rank_loss: 1.0
    dist_loss: 1.0`


as well as these 

      lr_attention_layer: 5.0e-05
      lr_project_v: 0.0
      lr_project_t: 0.0
      lr_project: 0.0

**Comment by duyao-art** [comment]
I have rechecked the default configuration file, and it should be sufficient to reproduce the reported results, consistent with the setup you used. In that sense, using default.yaml should be acceptable for reproduction.

One additional aspect that may be worth double-checking is the data split file. In our experiments, we used
datasets/FileList_1_shot_mediam.csv as the data file. The term “mediam” here only refers to a reduced validation set size, which we used to accelerate the validation process.

I will also try to reproduce the experiment on my side ASAP to verify whether there are any issues. At the moment, I am running another experiment and temporarily lack available computational resources, but I will follow up once resources are freed.

Based on the logs you shared, the model appears not to converge at all, which is indeed unexpected and worth further investigation.

Regarding the following parameters:  
rank_loss: 1.0  dist_loss: 1.0 lr_project_v: 0.0 lr_project_t: 0.0 lr_project: 0.0
these were part of experiments from a different project and are not related to the experiments reported in this paper.
I apologize for the confusion caused by my earlier reply

**Comment by AhmedAtefAhmedAly** [comment]
Okay, when you get a chance to reproduce it, let me know. 

thanks again for your response

**Comment by duyao-art** [comment]
Hi, we re-ran the model using the default yaml configuration, and the validation loss decreases as expected.
We have uploaded the corresponding log to: results/MICCAI/few-shot/1-shot/5e-5/run.log

Additionally, in the log you provided, we noticed that both "acc_exp_metric" and "acc_max_metric" are reported as 0.0. However, in echoclip/runner/runner.py, specifically in:  
def compute_per_example_metrics(self, logits, predictions, y, gather_type="exp"):
    return {f"mae_{gather_type}_metric": mae, "predict_y": predict_y_label, "GT": y}  
only the MAE metric is returned. Accuracy metrics are not computed in this function.
This is not a critical issue, but it may indicate a discrepancy in the implementation or logging pipeline on your side.

Could you please recheck your implementation and verify whether your log is consistent with: results/MICCAI/few-shot/1-shot/5e-5/run.log
